### PR TITLE
Report GetWFExecutionHistory long poll under custom operation tag

### PR DESCRIPTION
**Pull request by Ardagan**
**What changed?**
Long poll on GetWFExecutionHistory will be reported under "LongPollGetWorkflowExecutionHistory" operation.

**Why?**
Long poll generally has longer overall latency and false triggers alerts.

**How did you test it?**
Manual testing, automated tests.

**Potential risks**
Alerts on GetWorkflowExecutionHistory should be split into two, if you track long poll use case.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: common/metrics/defs.go [hunk 1]
```diff
@@ -743,8 +743,10 @@ const (
 	FrontendRespondActivityTaskFailedByIdScope
 	// FrontendRespondActivityTaskCanceledScope is the metric scope for frontend.RespondActivityTaskCanceledById
 	FrontendRespondActivityTaskCanceledByIdScope
-	// FrontendGetWorkflowExecutionHistoryScope is the metric scope for frontend.GetWorkflowExecutionHistory
+	// FrontendGetWorkflowExecutionHistoryScope is the metric scope for non-long-poll frontend.GetWorkflowExecutionHistory
 	FrontendGetWorkflowExecutionHistoryScope
+	// FrontendGetWorkflowExecutionHistoryScope is the metric scope for long poll case of frontend.GetWorkflowExecutionHistory
+	FrontendLongPollGetWorkflowExecutionHistoryScope
```
Comment (wxing1292) [inline common/metrics/defs.go:749]: `LongPollGet` let us talk about this name during the standup

File: common/rpc/interceptor/telemetry.go [hunk 2]
```diff
@@ -67,6 +68,22 @@ func NewTelemetryInterceptor(
...
 }
 
+
+// Use this method to override scope used for reporting a metric.
+// Ideally this method should never be used.
+func (ti *TelemetryInterceptor) overrideScope(scope int, methodName string, req interface{}) int {
+	// GetWorkflowExecutionHistory method handles both long poll and regular calls.
+	// Current plan is to eventually split GetWorkflowExecutionHistory into two APIs,
+	// remove this if case when that is done.
+	if scope == metrics.FrontendGetWorkflowExecutionHistoryScope && methodName == "GetWorkflowExecutionHistory" {
```
Comment (wxing1292) [inline common/rpc/interceptor/telemetry.go:78]: this check `scope == metrics.FrontendGetWorkflowExecutionHistoryScope` is unnecessary

Comment (Ardagan) [reply common/rpc/interceptor/telemetry.go:78]: It is not explicitly required, but I prefer to add it for safety in case of surrounding logic change.

Comment (wxing1292) [reply common/rpc/interceptor/telemetry.go:78]: then remove `methodName == "GetWorkflowExecutionHistory"`?

Comment (wxing1292) [reply common/rpc/interceptor/telemetry.go:78]: the mapping is already been unit tested
https://github.com/temporalio/temporal/blob/master/common/metrics/api_metrics_test.go

Comment (Ardagan) [reply common/rpc/interceptor/telemetry.go]: done.

File: common/metrics/defs.go (hunk 1 above)
Comment (alexshtin) [reply common/metrics/defs.go:749]: I believe `FrontendPollWorkflowExecutionHistoryScope` is the right name.

**Review by alexshtin — APPROVED** [review]

**Pull request merged**